molgenis-data-platform/src/main/java/org/molgenis/data/platform/decorators/MolgenisRepositoryDecoratorFactory.java

```diff
@@ -80,23 +84,26 @@ public MolgenisRepositoryDecoratorFactory(EntityManager entityManager,
 	{
 		Repository<Entity> decoratedRepository = repository;
 
-		// 14. Query the L2 cache before querying the database
+		// 15. Query the L2 cache before querying the database
 		decoratedRepository = new L2CacheRepositoryDecorator(decoratedRepository, l2Cache, transactionInformation);
 
-		// 13. Query the L1 cache before querying the database
+		// 14. Query the L1 cache before querying the database
 		decoratedRepository = new L1CacheRepositoryDecorator(decoratedRepository, l1Cache);
 
-		// 12. Route specific queries to the index
+		// 13. Route specific queries to the index
 		decoratedRepository = indexedRepositoryDecoratorFactory.create(decoratedRepository);
 
-		// 11. Query the L3 cache before querying the index
+		// 12. Query the L3 cache before querying the index
 		decoratedRepository = new L3CacheRepositoryDecorator(decoratedRepository, l3Cache, transactionInformation);
 
-		// 10. Register the cud action needed to index indexed repositories
+		// 11. Register the cud action needed to index indexed repositories
 		decoratedRepository = new IndexActionRepositoryDecorator(decoratedRepository, indexActionRegisterService);
 
-		// 9. Custom decorators
-		decoratedRepository = repositoryDecoratorRegistry.decorate(decoratedRepository);
+		// 10. Dynamic decorators
+		decoratedRepository = dynamicRepositoryDecoratorRegistry.decorate(decoratedRepository);
```

Comment:
Could you explain why the location of the dynamic decorators in the chain? Did you consider the security impact e.g. in case of an `evil` decorator?

Reply:
Still under consideration. At least the system entity type specific decorators should go first because of the row level security on metadata.